### DISCUSSION TITLE
How to debug monorepo with: Bundle group cannot have more than one entry bundle of the same type

### DISCUSSION THREAD

**Discussion by lindskogen** [parcel-bundler/parcel#5074] [General]
I downloaded the latest beta (`2.0.0-beta.1`).

We are using svg-as-react-component-imports so I needed to add a `.parcelrc` file:

```
{
  "extends": ["@parcel/config-default"],
  "transformers": {
    "*.svg": ["@parcel/transformer-svg-react"]
  }
}
```

Now I get this when I run `yarn parcel packages/*`.

```
@parcel/namer-default: Bundle group cannot have more than one entry bundle of the same type
AssertionError [ERR_ASSERTION]: Bundle group cannot have more than one entry bundle of the same type
```

How do I begin debugging this? Can I somehow get the original filenames of the conflicting bundles if I put a well-placed console.log somewhere?

Every package.json specifies their own types, module, source like so:

```
  "types": "dist/index.d.ts",
  "module": "dist/index.es.js",
  "source": "src/index.ts",
```

**Comment by mischnic** [comment]
Maybe logging `entryBundlesOfType` here (`console.log(entryBundlesOfType.map(b => b.filePath));`)?

https://github.com/parcel-bundler/parcel/blob/c4e494f3b9fd0eac672edc1a27c6c9216182dd6b/packages/namers/default/src/DefaultNamer.js#L26-L35

In the published npm package, it's in the `lib` folder:

https://unpkg.com/browse/@parcel/namer-default@2.0.0-beta.1/lib/DefaultNamer.js

**Reply by lindskogen** [reply]
I seem to get a log statement at the correct time, but I only get back `[null, null]`

**Reply by lindskogen** [reply]
I can log their type, which seems to be `css` both of them!

**Reply by mischnic** [reply]
The thing is: if the other bundles weren't named yet, well, you don't know their names/filepaths.
- Maybe you could try to figure something out via `getParentBundles`
- Try to reduce the `yarn parcel packages/*` glob so that its easier to reason about

**Reply by lindskogen** [reply]
It doesn't happen when I tried running parcel for every dir in packages:

```
find packages -type d -depth 1 -exec yarn parcel {} \;
```

**Reply by lindskogen** [reply]
It seems to happen when I build two packages A and B, when A depends on B and B has css files.

**Reply by TobiasSchikora** [reply]
Anything new? I have the same problems with parcel 2. See https://github.com/parcel-bundler/parcel/issues/6310.

**Reply by TobiasSchikora** [reply]
Nothing new? Problem still exists. See quick demo at https://github.com/parcel-bundler/parcel/issues/6310

**Reply by richardkmichael** [reply]
@mischnic I'm debugging this problem now too.  Thanks for the tip to `console.log` in `DefaultNamer.js`.

I have found `parcel serve` is generating `dist/index.css` twice.  It doesn't actually write `dist/` though, so I'm unsure if the content of the two generated files is is different.  I presume this data is in the LMDB cache, so I'll have a look there for hints about how the CSS is being generated.

FWIW, `parcel build ..` doesn't have this problem because it doesn't seem to be generating any CSS.  I don't know why.

I have six CSS import lines in the entire project, i.e., `import ...`.

- One CSS file imported in three different files
- A second CSS file imported in two different files
- A third CSS file imported in one file only

**Comment by S-unya** [comment]
I may have a little bit of new information here. I have this same problem, but **if I remove either the** `source` entry or the `module` entry from `package.json`, it works fine. My guess is that there are a number of assets that will be duplicated for the 2 output processes - in mine and @lindskogen 's case I think it is the svg assets.

I don't know of a work around for this apart from running parcel once for each output target, which seems inelegant.
